Tools/clinic/libclinic/converter.py

```diff
@@ -296,7 +296,10 @@ def _render_non_self(
             data.post_parsing.append('/* Post parse cleanup for ' + name + ' */\n' + post_parsing.rstrip() + '\n')
 
         # cleanup
-        cleanup = self.cleanup()
+        if parameter.is_vararg():
+            cleanup = f"Py_XDECREF({self.parser_name});"
+        else:
+            cleanup = self.cleanup()
         if cleanup:
             data.cleanup.append('/* Cleanup for ' + name + ' */\n' + cleanup.rstrip() + "\n")
```

Comment:
```suggestion
        if parameter.is_vararg():
            data.cleanup.append(f"Py_XDECREF({self.parser_name});")
        else:
            cleanup = self.cleanup()
            if cleanup:
                data.cleanup.append('/* Cleanup for ' + name + ' */\n' + cleanup.rstrip() + "\n")
```

Reply:
> Why? If this to remove the "`/* Cleanup for args */`" comment, I think that adding such comment is a good thing. It unifies the code with the code for other parameters.

For `Py_XDECREF(__clinic_args)` the comment does not add any value.

OTOH, cleanup for _parameters_ is more of a special case; they handle cases such as freeing memory using the `PyMem` APIs, or releasing buffer resources.

> Initially I planned to introduce a special converter for var-positional parameters, but it required some dirty hacks, because it needs access to some values like `max_pos`. I still may implement it in future to support representation of the var-positional parameter as a C array rather than a Python tuple, but this is a different issue. And we may implement support of var-keyword parameters.

That sounds really neat!

> So more uniform cleanup code is a good thing.

Well, it is a minor issue; I will not push further on this.

Reply:
The temporal special case has been removed, and the cleanup code for var-positional parameter is now generated in the `cleanup()` method, as for other parameters. According to the initial plan.